### PR TITLE
boards: add esp32s3-pros3 support

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR provides the support for [ESP32 ProS3](https://esp32s3.com/pros3.html#home) board from Unexpected Maker.

This board doesn't have a USB-to-Serial chip on board. Therefore, USB Serial/JTAG is used for STDIO and the board is flashed via the USB Seral/JTAG interface by default.

### Testing procedure

Flashing `tests/shell` should work.

### Issues/PRs references

### CODE REVIEW

**Comment by benpicco** [comment]
Does `stdio_usb_serial_jtag` work on this one?

**Comment by gschorcht** [comment]
> Does stdio_usb_serial_jtag work on this one?

Yes. So it seems to be better to use `stdio_usb_serial_jtag` by default to let the application the option to use USBUS or tinyUSB. Was that the background of your question?

**Comment by benpicco** [comment]
Yes because it should only require a single USB cable then, right? Because Serial/JTAG is a dedicated USB port and device USB would be on another one.

So if the user only has to connect to a single port for flashing and the shell, that would be more convenient.

**Comment by gschorcht** [comment]
> **edit** oh I see the board only has a single USB.
> How does that work? Can Serial/JTAG and device USB be switched at run-time?

Yeah, that's a bit tricky :wink: The ESP32S3 have two different controllers but a logic which one is routed to the D+/D- pin, see [ESP32S3 TRM](https://www.espressif.com/sites/default/files/documentation/esp32-s3_technical_reference_manual_en.pdf) page 1204. On reset, the USB Serial/JTAG interface is selected by default so that you can flash and use the CDC ACM interface. Only the USB OTG is initialized, the USB OTG signals are selected and the USB Serial/JTAG interface is not available any longer.

**Comment by benpicco** [comment]
That's interesting.
Yea I think `stdio_usb_serial_jtag` would be a good default since it *should* be more robust and light-weight as it acts more like a USB-UART adapter than a full USB stack.

But I'm mainly asking if it works on ESP32-S3 at all, I only received one in the mail today. 

Btw: Why is this still WIP?

**Comment by gschorcht** [comment]
Work on this board definition has been finished.

**Review by benpicco — APPROVED** [review]

**Comment by gschorcht** [comment]
@benpicco I had to provide some fixes due to compilation error in Murdock:

- The module `usb_board_reset` is also enabled when `stdio_usb_serial_jtag` is used as STDIO backend. Even if another STDIO backend like `stdio_uart` is flashed, the USB Serial/JTAG CDC ACM and thus a `ttyACMx` should be available. Although `usb_board_reset` doesn't affect the USB Serial/JTAG CDC ACM, it is necessary to be able to reset the board in download mode for flashing if `stdio_cdc_acm` or `stdio_tinyusb_cdc_acm` was previously flashed and is still active. Otherwise flashing via the USB Serial/JTAG CDC ACM interface is not possible. Commit 9f8b1e0d5dfbccd17bc1be5f67f66fde93adb5b8 allows to enable `usb_board_reset` in Kconfig also if only `stdio_usb_serial_jtag` is used.
- In PR #19128, a delay for tests was defined in `stdio_cdc_acm.dep.mk` to take into account the fact that it may take a few seconds before the CDC ACM serial device `ttyACM` becomes available. Commit 9810617889af73a7d30b64b40890db4498ad1b9e adds this change to `stdio_tinyusb_cdc_acm.dep.mk`.
- Commit 97373d821df3c29d663419d79eacc0f052375e23 adds a comment for `usb_board_reset` and enables it in Kconfig.

**Review by benpicco — APPROVED** [review]
Looks good to me - please squash!

**Comment by benpicco** [comment]
bors merge

File: boards/esp32s3-pros3/Makefile.features [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# the board uses a ESP32-S3 with a 16MB QSPI Flash and a 8MB QSPI PSRAM
+CPU_MODEL = esp32s3
+FEATURES_PROVIDED += esp_spi_ram
```
Comment (benpicco) [inline boards/esp32s3-pros3/Makefile.features:3]: This also needs to be provided in Kconfig

Comment (gschorcht) [reply boards/esp32s3-pros3/Makefile.features:3]: Did I forget it ? :thinking:

Comment (gschorcht) [reply boards/esp32s3-pros3/Makefile.features:3]: May I sqash it directly?

Comment (benpicco) [reply boards/esp32s3-pros3/Makefile.features:3]: Of course!

Comment (gschorcht) [reply boards/esp32s3-pros3/Makefile.features:3]: Done

**Comment by benpicco** [comment]
bors merge

**Comment by gschorcht** [comment]
Thanks for reviewing and merging.

**Pull request merged**